src/com/epam/Exceptions/helpers/UniversityHelper.java

```diff
@@ -0,0 +1,248 @@
+
+        Faculty faculty1 = new Faculty("Applied Mathematics", new Group[]{group1, group2});
+        Faculty faculty2 = new Faculty("romanGermanic", new Group[]{group3, group4});
+
+        return new University("YSU", "Alek Manukyan 1", new Faculty[]{faculty1, faculty2});
+    }
+
+    public static void calculateStudentsAverageMarkFromAllSubjects(Student student) {
+        float averageValue = 0;
+        if (student.getSubjects() == null || student.getSubjects().length == 0) {
+            throw new SubjectAbsenceException(student);
+        }
+        for (Mark mark : student.getMarks()) {
+            averageValue += mark.getValue();
+        }
+        System.out.println(student.getFirstName() + " " + student.getLastName() + " - average mark from all subjects : "
+                + averageValue / student.getMarks().length);
+    }
+
+    public static void calculateAverageMarkOfGroupFromSubject(Group group, String subject) {
+        float averageValue = 0;
+        int quantityOfMarks = 0;
+        if (group.getStudents() == null || group.getStudents().length == 0) {
+            throw new StudentAbsenceException(group);
+        }
+        for (Student student : group.getStudents()) {
+            for (Mark mark : student.getMarks()) {
+                if (mark.getSubject().equals(subject)) {
+                    averageValue += mark.getValue();
+                    ++quantityOfMarks;
+                }
+            }
+        }
+        System.out.println("Group : " + group.getName() + " - average mark from " + subject + " : "
+                + averageValue / quantityOfMarks);
+    }
+
+    public static void calculateAverageMarkOfFacultyFromSubject(Faculty faculty, String subject) {
+        float averageValue = 0;
+        int quantityOfMarks = 0;
+        if (faculty.getGroups() == null || faculty.getGroups().length == 0) {
+            throw new GroupAbsenceException(faculty);
+        }
+        for (Group group : faculty.getGroups()) {
+            if (group.getStudents() == null || group.getStudents().length == 0) {
+                throw new StudentAbsenceException(group);
+            }
+            for (Student student : group.getStudents()) {
+                for (Mark mark : student.getMarks()) {
+                    if (mark.getSubject().equals(subject)) {
+                        averageValue += mark.getValue();
+                        ++quantityOfMarks;
+                    }
+                }
+            }
+        }
+        System.out.println("Faculty : " + faculty.getName() + " - average mark from " + subject + " : "
+                + averageValue / quantityOfMarks);
+    }
+
+    public static void calculateAverageMarkOfUniversityFromSubject(University university, String subject)
+            throws FacultyAbsenceException {
+
+        float averageValue = 0;
+        int quantityOfMarks = 0;
+        if (university.getFaculties() == null || university.getFaculties().length == 0) {
```

Comment:
You may create little private helper methods like
checkUniversityFaculties or checkGroupStudents() to avoid code duplication in different methods.